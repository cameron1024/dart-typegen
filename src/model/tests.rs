use super::*;

macro_rules! parse_snapshot {
    ($name:ident) => {
        #[test]
        fn $name() {
            let text = include_str!(crate::test_file!($name));
            let config = Library::parse_impl(None, text).unwrap();

            insta::assert_debug_snapshot!(config);
        }
    };
}

crate::all_test_files!(parse_snapshot);
