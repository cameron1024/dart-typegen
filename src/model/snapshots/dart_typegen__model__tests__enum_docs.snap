---
source: src/model/tests.rs
expression: ron
---
Library {
    preamble: None,
    postamble: None,
    meta: None,
    defaults: None,
    enums: [
        Enum {
            span: Span(
                0,
                0,
            ),
            name: SpannedScalar {
                value: "Foo",
                span: SourceSpan {
                    offset: SourceOffset(
                        27,
                    ),
                    length: 5,
                },
            },
            docs: Some(
                SpannedScalar {
                    value: "\nMulti line comment\n\nMore stuff\n  ",
                    span: SourceSpan {
                        offset: SourceOffset(
                            42,
                        ),
                        length: 39,
                    },
                },
            ),
            annotations: None,
            variants: [
                EnumVariant {
                    name: SpannedScalar {
                        value: "a",
                        span: SourceSpan {
                            offset: SourceOffset(
                                93,
                            ),
                            length: 3,
                        },
                    },
                    docs: Some(
                        SpannedScalar {
                            value: "single line comment",
                            span: SourceSpan {
                                offset: SourceOffset(
                                    108,
                                ),
                                length: 21,
                            },
                        },
                    ),
                    json_value: None,
                },
                EnumVariant {
                    name: SpannedScalar {
                        value: "b",
                        span: SourceSpan {
                            offset: SourceOffset(
                                145,
                            ),
                            length: 3,
                        },
                    },
                    docs: None,
                    json_value: None,
                },
            ],
            extra_dart: [],
        },
    ],
    classes: [],
    unions: [],
}
